### PR TITLE
#167165066 Like or Dislike an Article

### PR DESCRIPTION
**Pull request by ocranbillions**
### What does this PR do?
- Enables likes or dislikes on articles

#### Description of Task to be completed?
- Add endpoints for like/dislike
- Get article slug from req.params
- Get userId from req.user
- Query db for article using the slug
- Get articleId from the query result
- Check if user has previously liked or disliked the article,
- If not, user can like or dislike article
- If yes user can remove like or remove dislike on the article

#### How should this be manually tested?
Liking an article:
> - Make a post request to `/articles/:slug/like` to like an article
  > - A second request removes the like

Disliking an article:
> - Make a post request to `/articles/:slug/dislike` to dislike an article
  > - A second request removes the dislike

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#167165066](https://www.pivotaltracker.com/story/show/167165066)

#### Screenshots (if appropriate)
Screenshot 2019-08-03 at 12 32 54 AM
Screenshot 2019-08-03 at 12 33 08 AM
Screenshot 2019-08-03 at 12 33 19 AM
Screenshot 2019-08-03 at 12 33 27 AM

### CODE REVIEW

File: controllers/LikesController.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+   * @param {function} next
+   * @returns {object} returns...
+   */
+  static async likeArticle(req, res, next) {
+    try {
+      const { slug } = req.params;
+      const userId = req.user.id;
+
+      const article = await models.Article.findOne({ where: { slug } });
+      if (!article) return notFoundError(req, res);
```
Comment (tolulope-od) [inline controllers/LikesController.js]: Hi could you please change use the `errorResponse` method so a more specific message can be sent back to the client?

Comment (ocranbillions) [reply controllers/LikesController.js]: Ok cool.

File: controllers/LikesController.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      });
+
+      // Update count on article table
+      await models.Article.update(
+        { likes: (article.dataValues.likes + 1) },
+        { where: { slug } },
+      );
+
+      return successResponse(
+        res, 200, 'message',
```
Comment (mazma1) [inline controllers/LikesController.js]: Shouldn't this be a `201`?

File: controllers/LikesController.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      });
+
+      // Update count on article table
+      await models.Article.update(
+        { dislikes: (article.dataValues.dislikes + 1) },
+        { where: { slug } },
+      );
+
+      return successResponse(
+        res, 200, 'message',
```
Comment (mazma1) [inline controllers/LikesController.js]: And this too, `201`?

File: controllers/LikesController.js (hunk 2 above)
Comment (ocranbillions) [reply controllers/LikesController.js]: Yes it should. I'll change it right away.

File: controllers/LikesController.js (hunk 3 above)
Comment (ocranbillions) [reply controllers/LikesController.js]: ok.

**Comment by mazma1** [comment]
Good work reviewing the implementation of this feature, @ocranbillions!

**Pull request merged**